### PR TITLE
Fixes #9773 - Add article about injections attacks

### LINKED ISSUE #9773: Document using EscapeSingleQuotedStringContent API to prevent script injection
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Document new `Get-Foo` cmdlet" instead of "New cmdlet."

### PowerShell Version

5.1, 7.2, 7.3, 7.4

### Summary

We should document the `EscapeSingleQuotedStringContent` API and how it should be used to prevent script injection with single quoted arguments.  We should also have a general article (blog?) about script injection and mention the 'Script injection hunter for PSScriptAnalyzer.

### Details

_No response_

### Proposed Content Type

About Topic

### Proposed Title

_No response_

### Related Articles

_No response_

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Add article about injections attacks

- Fixes [AB#62438](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/62438)
- Fixes #9773

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/docs-conceptual/dev-cross-plat/security/preventing-script-injection.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
+---
+description: This article explains how to prevent script injection attacks using single quote escaping.
+ms.date: 02/10/2023
+title: Preventing script injection attacks
+---
+# Preventing script injection attacks
+
+PowerShell scripts, like other programming language, can be vulnerable to injection attacks. An
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/dev-cross-plat/security/preventing-script-injection.md:8]:
```suggestion
PowerShell scripts, like other programming languages, can be vulnerable to injection attacks. An
```

File: reference/docs-conceptual/dev-cross-plat/security/preventing-script-injection.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+injection attack occurs when a user provides input to a vulnerable function that includes extra
+commands. The vulnerable function runs the extra commands, which is can be a serious security
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/dev-cross-plat/security/preventing-script-injection.md:10]:
```suggestion
commands. The vulnerable function runs the extra commands, which can be a serious security
```

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
Great write-up, only changes are for a few minor typos.

**Review by michaeltlombardi — DISMISSED** [review]

**Review by PaulHigin — APPROVED** [review]
LGTM.  Just a few comments.

File: reference/docs-conceptual/dev-cross-plat/security/preventing-script-injection.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+This example uses single quotes in the user input to force the function to run three separate
+statements, one of which is arbitrary code injected by the user.
+
+```Output
+ NPM(K)    PM(M)      WS(M)     CPU(s)      Id  SI ProcessName
+ ------    -----      -----     ------      --  -- -----------
+     97    46.08     183.10       1.08    2524   3 pwsh
+pwnd!
+```
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/docs-conceptual/dev-cross-plat/security/preventing-script-injection.md:159]: The sample works as written for me.

**Pull request merged**